tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -137,6 +137,13 @@ DuckDBPyConnection *DuckDBPyConnection::Execute(const string &query, py::object
 	result = nullptr;
 	unique_ptr<PreparedStatement> prep;
 	{
+		// we first release the gil and then acquire the connection lock
+		unique_lock<std::mutex> lock(py_connection_lock, std::defer_lock);
```

Comment:
Would it be possible to make this a function, e.g.: `unique_lock<std::mutex> DuckDBPyConnection ::AcquireLock()`?